src/core/trulens/core/app/base.py

```diff
@@ -91,8 +93,22 @@
 ```
 """
 
+_component_impls: Dict[str, Type[ComponentView]] = {}
+
+
+class ComponentViewMeta(ABCMeta):
```

Comment:
Logic change: using metaclasses to add anything that inherits from `ComponentView` to `_component_impls`. 

This replaces definitions from instrumentation extensions here (e.g. LangchainComponentView) here, which would have dependencies. Also should make it easier to add new instrumentation implementations